client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -656,6 +657,18 @@ static Request getRepositories(GetRepositoriesRequest getRepositoriesRequest) {
         return request;
     }
 
+    static Request putRepository(PutRepositoryRequest putRepositoryRequest) throws IOException {
+        String endpoint = new EndpointBuilder().addPathPart("_snapshot").addPathPart(putRepositoryRequest.name()).build();
+        Request request = new Request(HttpPut.METHOD_NAME, endpoint);
+
+        Params parameters = new Params(request);
+        parameters.withMasterTimeout(putRepositoryRequest.masterNodeTimeout());
+        parameters.withTimeout(putRepositoryRequest.timeout());
+
+        request.setEntity(createEntity(putRepositoryRequest, REQUEST_BODY_CONTENT_TYPE));
```

Comment:
I see in the corresponding REST action that we also support the `verify` parameter. I guess we should support it here too?

Reply:
yea definitely, ty for pointing it out. Cant believe i missed it!